packages/devextreme/js/__internal/core/license/license_validation.ts

```diff
@@ -162,8 +165,26 @@ export function validateLicense(licenseKey: string, versionStr: string = package
   }
   validationPerformed = true;
 
+  const dependentVersions = getDependentVersions();
+  const mismatchingDependents = dependentVersions.filter(
+    (dependent) => !versionsEqual(
+      dependent.version,
+      versionStr,
+    ),
+  );
+
+  if (mismatchingDependents.length) {
+    errors.log('W0023', stringifyVersionList([
+      {
+        dependentName: 'devextreme',
+        version: versionStr,
+      },
+      ...mismatchingDependents,
+    ]));
+    return;
+  }
```

Comment:
This should be a part of the version module. Version check and license check are routines of the same level and should be independent from each other

As for now, there are no other consumers for the version check output, so we can call it from within the license check, but license check should now nothing of it's implementation